addon/experiment-api/contactsApi.js

```diff
@@ -53,6 +100,61 @@ var convContacts = class extends ExtensionCommon.ExtensionAPI {
             args
           );
         },
+        async composeNew(properties) {
+          const window =
+            properties.windowId !== null
+              ? windowManager.get(properties.windowId, context).window
+              : Services.wm.getMostRecentWindow("mail:3pane");
+          const { to } = properties;
+          composeMessageTo(to, window.gFolderDisplay.displayedFolder);
+        },
+        async showMessagesInvolving(options) {
+          const window =
+            options.windowId !== null
+              ? windowManager.get(options.windowId, context).window
+              : Services.wm.getMostRecentWindow("mail:3pane");
+
+          const { name, email } = options;
+
+          const collection = await glodaQuery({
+            kind: "email",
+            value: email,
+            query: "NOUN_IDENTITY",
+          });
+          if (!collection.items.length) {
+            return;
+          }
+
+          const result = await glodaQuery({
+            query: "NOUN_MESSAGE",
+            involvesItems: collection.items,
+          });
+
+          const browser = BrowserSim.getBrowser();
+          let tabmail = window.document.getElementById("tabmail");
+          tabmail.openTab("glodaList", {
+            collection: result,
```

Comment:
Have think I have a way of making this open faster when doing large queries, but I'll look at that once this has landed.